assets/app/views/directives/osc-key-values.html

```diff
@@ -23,7 +25,15 @@
                  spellcheck="false"
                  osc-input-validator="value">
         </div>
-        <button class="btn btn-default" ng-click="addEntry()" ng-disabled="form.key.$error.oscKeyValid || form.value.$error.oscValueValid">Add</button>
+        <button
+          class="btn btn-default"
+          ng-click="addEntry()"
+          ng-disabled="form.key.$error.oscKeyValid || form.value.$error.oscValueValid || form.key.$error.oscUnique">
```

Comment:
Aha, had to remove `form.$setValidity()` from `$scope.addEntry()`, then using `ng-disabled="form.$invalid"` works.
Maybe was overkill on trying to reset form state.